lib-object-oriented/line.js

```diff
@@ -1,34 +1,77 @@
 import Tag from './tag'
 
 export default class Line {
-  constructor( content ) {
+  constructor({ content, type }) {
     this.content = content
+    this.tagType = type
+    this.symbolCount = 0
+    this.getSymbolCount()
+    this.newLinesToSpaces()
+  }
+
+  newLinesToSpaces() {
+    const tagType = this.tagType,
+          content = this.content,
+          symbolCount = this.symbolCount
+
+    const findNewLine = element => {
+      return element.type === 'new_line'
+    }
+
+    if ( tagType !== '`' || ( tagType === '`' && symbolCount === 1 ) ) {
+      while ( content.findIndex( findNewLine ) > -1 ) {
+        const newLine = content.find( findNewLine )
+        if ( newLine ) {
+          newLine.type = ' '
+          newLine.value = ' '
+        }
+      }
+    }
+  }
+
+  getSymbolCount() {
+    for ( let index = 0; index < this.content.length; index++ ) {
+      const currentToken = this.content[ index ]
+      if ( currentToken.type.search( /([#>`])/g) !== -1 ) {
+        if ( this.tagType === currentToken.type && index <= this.symbolCount ) {
+          this.symbolCount++
+        }
+      }
+    }
   }
 
   parse() {
     let output = '',
-        symbolCount = 0,
-        tagType
+        tagType = this.tagType
 
-    this.content.forEach( token => {
-      if ( token.type === '#' ) {
-        tagType = token.type
-        symbolCount++
+    for ( let index = 0; index < this.content.length; index++ ) {
+      const currentToken = this.content[ index ]
+
+      if ( currentToken.type.search( /([#>`])/g) !== -1 ) {
+        if ( index === 0 ) { tagType = currentToken.type }
+        if ( tagType === '`' && currentToken.type !== tagType ) {
+          output += currentToken.value
+        }
+        if ( currentToken === '`' ) {
```

Comment:
Another opportunity (or at least a re-use of the constant declared in tokens.js)